internal/provisioner/teleport.go

```diff
@@ -88,15 +91,35 @@ func (n *teleport) ActualVersion() string {
 }
 
 func (n *teleport) Destroy() error {
+	teleportClusterName := fmt.Sprintf("cloud-%s-%s", n.environment, n.cluster.ID)
+	err := n.awsClient.S3EnsureBucketDeleted(teleportClusterName, n.logger)
```

Comment:
Where does this bucket come from? I took a quick look and didn't see anything related to creating this utility.

Reply:
@gabrieljackson your question makes sense. Both the S3 bucket and the DynamoDB tables are automatically created by the Teleport helm chart deployment.

Reply:
Oh, that makes sense! Thanks.

I am guessing we can't have helm tear them down for us as well?

Reply:
Nope unfortunately it cannot. I guess the reason for that is that they don't want to delete any data we may want to preserve in an accidental helm chart deletion.